server/rpc/interceptors/interceptors.go

```diff
@@ -368,10 +371,13 @@ func quotaStreamServerInterceptor(env environment.Env) grpc.StreamServerIntercep
 			if err != nil {
 				log.Warningf("Quota Manager failed: %s", err)
 			}
-		}
-		if *enableGRPCMetricsByGroupID {
-			if key, err := quota.GetKey(stream.Context(), env); err == nil {
-				metrics.RPCsHandledTotalByQuotaKey.WithLabelValues(info.FullMethod, key, strconv.FormatBool(allow)).Inc()
+			if !allow {
+				return status.ResourceExhaustedError("exceeds quota")
+			}
+			if *enableGRPCMetricsByGroupID {
+				if key, err := quota.GetKey(stream.Context(), env); err == nil {
+					metrics.RPCsHandledTotalByQuotaKey.WithLabelValues(info.FullMethod, key, strconv.FormatBool(allow)).Inc()
+				}
```

Comment:
Can we have the quota manager handle these metrics internally? Also, feels like the metric should be more generic, using the "namespace" concept for the metric, instead of an RPC specific metric.